docs/proposals/volumes.md

```diff
@@ -0,0 +1,431 @@
+the following scenarios:
+    1.  All containers running as root
+    4.  All containers running as the same non-root user
+    5.  Multiple containers running as a mix of root and non-root users
+
+### All containers running as root
+
+For volumes that only need to be used by root, no action needs to be taken to change ownership or
+permissions, but setting the ownership based on the supplemental group shared by all containers in a
+pod will also work.  For situations where read-only access to a shared volume is required from one
+or more containers, the `VolumeMount`s in those containers should have the `readOnly` field set.
+
+### All containers running as a single non-root user
+
+In use cases whether a volume is used by a single non-root UID the volume ownership and permissions
+should be set to enable read/write access.
+
+Currently, a non-root UID will not have permissions to write to any but an `EmptyDir` volume.
+Today, users that need this case to work can:
+
+1.  Grant the container the necessary capabilities to `chown` and `chmod` the volume:
+    - `CAP_FOWNER`
+    - `CAP_CHOWN`
+    - `CAP_DAC_OVERRIDE`
+2.  Run a wrapper script that runs `chown` and `chmod` commands to set the desired ownership and
+    permissions on the volume before starting their main process
+
+This workaround has significant drawbacks:
+
+1.  It grants powerful kernel capabilities to the code in the image
+2.  The user experience is poor; it requires changing Dockerfile, adding a layer, or modifying the
+    container's command
+
+Some cluster operators manage the ownership of shared storage volumes on the server side.
+In this scenario, the UID of the container using the volume is known in advance.  The ownership of
+the volume is set to match the container's UID on the server side.
+
+### Containers running as a mix of root and non-root users
+
+If the list of UIDs that need to use a volume includes both root and non-root users, supplemental
+groups can be applied to enable sharing volumes between containers.  The ownership and permissions
+`root:<supplemental group> 0770` will make a volume usable from both containers running as root and
+running as a non-root UID and the supplemental group.
+
+## Community Design Discussion
+
+- [kubernetes/2630](https://github.com/GoogleCloudPlatform/kubernetes/issues/2630)
+- [kubernetes/11319](https://github.com/GoogleCloudPlatform/kubernetes/issues/11319)
+- [kubernetes/9384](https://github.com/GoogleCloudPlatform/kubernetes/pull/9384)
+
+## Analysis
+
+The system needs to be able to:
+
+1.  Model correctly which volumes require ownership management
+1.  Determine the correct ownership of each volume in a pod if required
+1.  Set the ownership and permissions on volumes when required
+
+### Modeling whether a volume requires ownership/label management
+
+#### Unshared storage: volumes derived from `EmptyDir`
+
+The `EmptyDir` plugin and volumes derived from it are the opposite of hostPath.  Since they are
```

Comment:
This first sentence is unnecessary.  `Since Kubernetes creates emptyDir volumes, it should ensure the ownership is set to enable the volume to be usable for all of the above scenarios`